diff --git a/kernel/supercalls.c b/kernel/supercalls.c
index de0731063fbd2fd3b1dad77ae7a4652ca38acbe4..2a7090f6b89ca7e41bea29deebf12f11c854bb1d 100644
--- a/kernel/supercalls.c
+++ b/kernel/supercalls.c
@@ -892,8 +892,8 @@ int ksu_handle_sys_reboot(int magic1, int magic2, unsigned int cmd,
 #endif //#ifdef CONFIG_KSU_SUSFS_SUS_PATH
 #ifdef CONFIG_KSU_SUSFS_SUS_MOUNT
-        if (cmd == CMD_SUSFS_HIDE_SUS_MNTS_FOR_ALL_PROCS) {
-            susfs_set_hide_sus_mnts_for_all_procs(arg);
+        if (cmd == CMD_SUSFS_HIDE_SUS_MNTS_FOR_NON_SU_PROCS) {
+            susfs_set_hide_sus_mnts_for_non_su_procs(arg);
             return 0;
         }
 #endif //#ifdef CONFIG_KSU_SUSFS_SUS_MOUNT