diff --git a/kernel/Kbuild b/kernel/Kbuild
index 44d16b7b..2d1703c3 100644
--- a/kernel/Kbuild
+++ b/kernel/Kbuild
@@ -89,6 +89,10 @@ endif
 ccflags-y += -DKSU_VERSION=$(KSU_VERSION)
 ccflags-y += -DKSU_VERSION_FULL=\"$(KSU_VERSION_FULL)\"
 
+ifeq ($(CONFIG_KSU_MULTI_MANAGER_SUPPORT), y)
+$(info -- Supported Unofficial Manager: MKSU, RKSU, SukiSU-Ultra, ReSukiSU )
+endif
+
 # Checks hooks state
 ifeq ($(CONFIG_KSU_MANUAL_HOOK), y)
 $(info -- KernelSU: CONFIG_KSU_MANUAL_HOOK)
diff --git a/kernel/Kconfig b/kernel/Kconfig
index 8198f408..d689e1db 100644
--- a/kernel/Kconfig
+++ b/kernel/Kconfig
@@ -15,6 +15,13 @@ config KSU_DEBUG
 	help
 	  Enable KernelSU debug mode.
 
+config KSU_MULTI_MANAGER_SUPPORT
+	bool "Multi KernelSU manager support"
+	depends on KSU
+	default n
+	help
+		Enable multi KernelSU manager support
+		
 config KSU_ALLOWLIST_WORKAROUND
 	bool "KernelSU allowlist workaround"
 	depends on KSU
diff --git a/kernel/manager_sign.h b/kernel/manager_sign.h
index f9010c40..ada41393 100644
--- a/kernel/manager_sign.h
+++ b/kernel/manager_sign.h
@@ -33,6 +33,11 @@
 #define EXPECTED_HASH_NEKO                                                     \
 	"946b0557e450a6430a0ba6b6bccee5bc12953ec8735d55e26139b0ec12303b21"
 
+// ReSukiSU/ReSukiSU
+#define EXPECTED_SIZE_RESUKISU 0x377
+#define EXPECTED_HASH_RESUKISU                                                 \
+    "d3469712b6214462764a1d8d3e5cbe1d6819a0b629791b9f4101867821f1df64"
+	
 typedef struct {
 	u32 size;
 	const char *sha256;
