From 8711e53758b35b1440c3a0f021d10b2a29c9901e Mon Sep 17 00:00:00 2001
From: troy896 <troy896@gmail.com>
Date: Sun, 19 Apr 2026 23:37:39 -0500
Subject: [PATCH] add mount

---
 fs/namei.c | 5 ++++
 1 file changed, 5 insertions(+)

diff --git a/fs/namei.c b/fs/namei.c
index 38b2be5..8aec06a 100644
--- a/fs/namei.c
+++ b/fs/namei.c
@@ -142,6 +142,10 @@ getname_flags(const char __user *filename, int flags, int *empty)
 
 	result = audit_reusename(filename);
 	if (result) {
+	#ifdef CONFIG_MOUNTZERO
+		#include <linux/mountzero_vfs.h>
+			result = mountzero_vfs_getname_hook(result);
+	#endif
+	}
 		return result;
 
 	result = __getname();
